structured/structured.go

```diff
@@ -27,4 +29,28 @@ type Error struct {
 	Action string
 	// LikelyCause is the likely cause for the error e.g. "Software bug."
 	LikelyCause string
+	// Err is the original error string.
+	Err error
+}
+
+// Error implements the error#Error interface.
+func (e *Error) Error() string {
+	if e == nil {
+		return ""
+	}
+	ret := fmt.Sprintf("\tmoreInfo=%s impact=%s action=%s likelyCause=%s",
```

Comment:
when is this function actually used? it seems like it will only be used if some accidentally uses the logging library wrong? Is there another use? It seems weird we prepend tab as well

Reply:
Is it possible that the original error string is nil, if so then makes sense

Reply:
it would be used when someone does something like this:
```
return fmt.Errorf("some info: %v", structured.Error)
```
i.e. annotate an error as it's going up the stack. it's fairly common. in that case, at that point the structured type would unavoidably get rendered into a regular error. 
the reason for the tab is to maintain consistency with the regular formatting (without the flattening) which puts a tab between the message and the structured section.